SwissTransferTests/SwissTransferTests.swift

```diff
@@ -17,6 +17,46 @@
  */
 
 import Foundation
+@testable import STNewTransferView
 import XCTest
 
-final class SwissTransferTests: XCTestCase {}
+@MainActor
+final class SwissTransferTests: XCTestCase {
+    func testDestinationURL() {
+        do {
+            // GIVEN
+            let newTransferManager = NewTransferManager()
+            let fileName = "my-file.txt"
+            let sourcePath = URL(string: "http://my-url.com/\(fileName)")!
+            let expectedResult = try URL.tmpUploadDirectory().appendingPathComponent(fileName)
+
+            // WHEN
+            let result = try newTransferManager.destinationURLFor(source: sourcePath)
+
+            // THEN
+            XCTAssertEqual(result, expectedResult)
+        } catch {
+            XCTAssert(false, "Error creating destination URL")
+        }
+    }
+
+    func testRenameDestinationURL() {
+        do {
+            // GIVEN
+            let newTransferManager = NewTransferManager()
+            let sourcePath = URL(string: "http://my-url.com/my-file.txt")!
+            let tmpDirectory = try URL.tmpUploadDirectory()
+            let expectedResult = tmpDirectory.appendingPathComponent("my-file(1).txt")
+            let firstURL = tmpDirectory.appendingPathComponent("my-file.txt", conformingTo: .text)
+            FileManager.default.createFile(atPath: firstURL.path(), contents: nil)
+
+            // WHEN
+            let result = try newTransferManager.destinationURLFor(source: sourcePath)
```

Comment:
It looks like you rely on `testDestinationURL` running before this test to create a conflict. This is relatively unsafe and can create flaky tests.
If a file should be added to the filesystem, you should do it in the `// GIVEN` part of the test. The clean-up can be performed in the `teardown` class method.